spec/services/discouse-calendar/holiday_spec.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module DiscourseCalendar
+  describe Holiday do
+    describe ".find_holidays_for" do
+      before do
+        DisabledHoliday.create!(holiday_name: "New Year's Day", region_code: "sg")
+        DisabledHoliday.create!(holiday_name: "Chinese New Year", region_code: "sg")
+      end
+
+      let(:holidays) do
+        Holiday.find_holidays_for(
+          region_code: "sg",
+          start_date: "2022-01-01",
+          end_date: "2022-04-30"
+        )
+      end
+
+      it "returns a list of holidays indicating whether a holiday is disabled or not" do
+        expect(holidays).to include(a_hash_including(
+          { name: "New Year's Day", regions: [:sg], disabled: true }
+        ))
+
+        expect(holidays).to include(a_hash_including(
+          { name: "Chinese New Year", regions: [:sg], disabled: true }
+        ))
+
+        expect(holidays).to include(a_hash_including(
+          { name: "Good Friday", regions: [:sg], disabled: false }
+        ))
+      end
+
+      describe "dates holidays are observed on" do
+        let(:holidays) { Holiday.find_holidays_for(
+          region_code: "sg",
+          start_date: "2021-12-31",
+          end_date: "2022-05-31",
+          show_holiday_observed_on_dates: show_holiday_observed_on_dates) }
+
```

Comment:
I added the test cases below for returning the dates holidays are observed on. Just a random FYI about this (in case it's not clear/known), the observed on dates get added to the calendar, but the admin shows the actual holiday dates, not the observed on dates. Which is why I tweaked the explanatory text to remove "scheduled" (as in "scheduled holidays"), because I didn't want it to cause confusion that the dates appearing in the admin wouldn't necessarily be the same dates that would get removed from the calendar, if that makes sense! :sweat_smile: 

And another side note, at first glance I don't think the Holidays gem can return both dates, but it could be done by making two calls to the gem - if we think it would be useful to display both dates in the admin. :smile: